client/web/compose/src/components/ModuleFields/Editor/Record.vue

```diff
@@ -297,7 +295,7 @@ export default {
     getRecord (index = undefined) {
       const value = index !== undefined ? this.value[index] : this.value
       if (value) {
-        return this.convert({ recordID: value })
+        return this.convert({ recordID: value }).value
```

Comment:
Whenever you do a .value, you need to be sure that what you're reading from is always an object and can never be undefined